mapflow/functional/layer_utils.py

```diff
@@ -370,6 +384,33 @@ def download_results_file_callback(self, response: QNetworkReply, path: str) ->
             f.write(response.readAll().data())
         self.message_bar.pushSuccess(self.tr("Results saved"),
                                      self.tr("see in {path}!").format(path=path))
+        
+    def download_aoi_file_callback(self, response: QNetworkReply, path: str) -> None:
+        """
+        Write area of interest to the geojson file
+        """
+        self.dlg.saveResultsButton.setEnabled(True)
+        with open(path, mode='wb') as f:
+            data = f.write(response.readAll().data())
+        with open(path) as f:
+            data = json.load(f)
+            geojson = {
+                "type": "FeatureCollection",
+                "features": [{
+                    "type": "Feature",
+                    "geometry" : d['geometry'],
+                    "properties" : {"id": d['id'], 
+                                    "status": d['status'],
+                                    "percentCompleted": d['percentCompleted'],
+                                    "area": d['area'],
+                                    "messages": d['messages']},
+                }for d in data]
+            }
+        with open(path, "w") as out:
+            json.dump(geojson, out)
+        aoi_layer = QgsVectorLayer(path, 'Area of interest')
```

Comment:
1. Use self.tr() on string literals so that we could translate them before release
2. Maybe add processing ID or something to identify the AOI between others?

Reply:
If this name is not to long, then okay
![image](https://github.com/Geoalert/mapflow-qgis/assets/136620754/7d088a5d-dab0-448a-9e9d-a8ff02f470fd)

Reply:
Maybe it should just say "AOI_{id}"?
Also, what about button naming? Should it stay "Save area"? Or should I change it to "Download AOI" or something?

Reply:
AOI: {id} is OK! (wiithout translation)
 We already use similar names for locally created layers